application/lib/service/user_database_service.dart

```diff
@@ -23,16 +23,18 @@ class UserDatabaseService {
             .collection('users')
             .doc(userCredential.user!.uid)
             .set(userDTO.toJson());
+        return UserDTO.fromJson(
```

Comment:
Itt az oda-visszakonvertálások helyett egyszerűbben is lehet. Kérlek csináljuk így: 
`userDTO.id = userCredential.user?.uid ?? ''
 return userDTO;`